tap_kafka/local_store.py

```diff
@@ -137,7 +137,13 @@ def insert(self, message: str) -> None:
         """Insert a new message with a generated timestamp
         Returns the last timestamp of the last inserted message"""
         # Add to the in-memory batch to avoid too frequent I/O write
-        self.messages_to_persist.append(message)
+        message_to_append = message
+        if isinstance(message, bytes):
```

Comment:
is this method `insert` called before or after the message has a been deserialized?

I assume the the kafka consumer would deserialize the message from bytes to dictionary upon reading from the brokers, the message is then turned into stringified singer message, so why does this method handle bytes

Reply:
bytes to string is required because orjson returns JSONs as bytes and not string. So, since `ppw-singer-python-2.0` is using orjson, then the singer JSON messages are generated as bytes and not string.

The extra decode is not required when writing directly to stdout, but it's required when appending into file in the local store. We'll get rid of the local store, so this piece of code will go away anyway soon.

Reply:
gotcha!